runtime-job-worker/src/test/java/io/camunda/connector/runtime/jobworker/FeelEngineWrapperTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.runtime.jobworker;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class FeelEngineWrapperTest {
+
+  // used to do test validation easier
+  private static final ObjectMapper UTILITY_JSON_MAPPER = new ObjectMapper();
+
+  private FeelEngineWrapper objectUnderTest;
+
+  @BeforeEach
+  void beforeEach() {
+    objectUnderTest = new FeelEngineWrapper();
+  }
+
+  @Test
+  void evaluateFeelExpressionToJson_ShouldSucceed_WhenHappyCase() throws JsonProcessingException {
+    // given
+    // FEEL expression -> {"processedOutput":response.callStatus}
+    final String resultExpression = "{\"processedOutput\": response.callStatus }";
+    // Response from service -> {"callStatus":{"statusCode":"200 OK"}}
+    final Map variables = Map.of("callStatus", Map.of("statusCode", "200 OK"));
+
+    // when
+    final String evaluatedResultAsJson =
+        objectUnderTest.evaluateFeelExpressionToJson(resultExpression, variables);
+
+    // then
+    Map<String, Object> jsonVars = UTILITY_JSON_MAPPER.readValue(evaluatedResultAsJson, Map.class);
+    assertThat(jsonVars.size()).isEqualTo(1);
+    assertThat(jsonVars).containsKey("processedOutput");
+    assertThat(jsonVars.get("processedOutput")).isInstanceOf(Map.class);
+    Map resultEntry = (Map) jsonVars.get("processedOutput");
+    assertThat(resultEntry).containsEntry("statusCode", "200 OK");
```

Comment:
How about we simplify those tests and compare the result against the expected JSON String?
```suggestion
    assertThat(evaluatedResultAsJson).isEqualTo("{\"processedOutput\":{\"statusCode\":\"200 OK\"}");
```

Reply:
That's not a bad idea, but, IMO, that causes too much pain when we do refactoring of respective code. What I see people doing in that case - once refactoring complete, people run tests and once it doesn't pass, they blindly copy-paste **expected** value.

So I would prefer to keep as object comparison but if still think we should compare JSON, I have now objections. @tmetzke WDYT?

Reply:
I see what you mean. The thing is, the expected result ideally shouldn't change when we adjust any code. I would prefer easy-to-read and therefore usually also more maintainable test cases. Potential future sloppy development practices shouldn't be an argument for keeping more complex code that's harder to change, I think. 🙂

Reply:
I brought the argument regarding developer being lazy because I saw production incidents related to that :). But okay, as I say, I don't have objections in this relation because I in my opinion both make sense to me thus I wil change this.